lib/trip_dispatcher.rb

```diff
@@ -87,7 +87,72 @@ def load_trips
         trips << trip
       end
 
-      trips
+      return trips
+    end
+
+    def request_trip(passenger_id)
+
+      if find_passenger(passenger_id).nil?
+        raise ArgumentError.new("A trip can only be requested for an existing customer.")
+      end
+
+      new_trip_data = {
+        id: new_trip_id,
+        passenger: find_passenger(passenger_id),
+        driver: select_driver,
+        start_time: Time.now,
+        end_time: nil,
+        cost: nil,
+        rating: nil
+      }
+
+      new_trip = Trip.new(new_trip_data)
+
+      @trips << new_trip
+      new_trip.driver.add_trip(new_trip)
+      new_trip.passenger.add_trip(new_trip)
+
+      new_trip.driver.change_status
+
+      return new_trip
+    end
+
+    def new_trip_id
+      @trips.map { |trip| trip.id }.max + 1
+    end
+
+    def available_drivers
+      available_drivers = @drivers.select do |driver|
+        driver.status == :AVAILABLE && driver.trips.length > 0
+      end
```

Comment:
Again, great use of helper methods in this file - these feel like they're just the right size and level of complexity.